node/src/Commands.ts

```diff
@@ -1961,6 +1973,124 @@ export function createGeoHash(
     return createCommand(RequestType.GeoHash, args);
 }
 
+// This type unites `GeoSearchResultOptions` and `GeoSearchOptions` of the java client
+/**
+ * Optional parameters for {@link BaseClient.geosearch|geosearch} command which defines what should be included in the
+ * search results and how results should be ordered and limited.
+ */
+export type GeoSearchResultOptions = {
+    /** Include the coordinate of the returned items. */
+    readonly withCoord?: boolean;
+    /**
+     * Include the distance of the returned items from the specified center point.
+     * The distance is returned in the same unit as specified for the `searchBy` argument.
+     */
+    readonly withDist?: boolean;
+    /** Include the geohash of the returned items. */
+    readonly withHash?: boolean;
+    /** Indicates the order the result should be sorted in. */
+    readonly sortOrder?: SortOrder;
+    /** Indicates the number of matches the result should be limited to. */
+    readonly count?: number;
+    /** Whether to allow returning as enough matches are found. This requires `count` parameter to be set. */
+    readonly isAny?: boolean;
+};
+
+/** Defines the sort order for nested results. */
+export enum SortOrder {
+    /** Sort by ascending order. */
+    ASC = "ASC",
+    /** Sort by descending order. */
+    DESC = "DESC",
+}
+
+export type GeoSearchShape = GeoCircleShape | GeoBoxShape;
+
+/** Circle search shape defined by the radius value and measurement unit. */
+export type GeoCircleShape = {
+    /** The radius to search by. */
+    readonly radius: number;
+    /** The measurement unit of the radius. */
+    readonly unit: GeoUnit;
+};
+
+/** Rectangle search shape defined by the width and height and measurement unit. */
+export type GeoBoxShape = {
+    /** The width of the rectangle to search by. */
+    readonly width: number;
```

Comment:
Should this be `readonly`? A user can save options to be passed around to different calls of the commands. With this change though we are restricting users.